### PR TITLE
fixed tests-2

### PR DESCRIPTION
**Pull request by skushagra9**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation

### Before this PR

{Please add a screenshot here}

### After this PR

{Please add a screenshot here}

### Related Issue (optional)

#623 
removed the 3 entry for statusReport as not needed
added the iso8601 validation to dates

### CODE REVIEW

**Comment by skushagra9** [comment]
adding for monitor

File: apps/server/src/v1/statusReportUpdate.ts [hunk 1]
```diff
@@ -37,10 +37,15 @@ export const statusUpdateSchema = z.object({
...
-    .preprocess((val) => String(val), z.string())
-    .openapi({
-      description: "The date of the update in ISO 8601 format",
-    }),
+  .preprocess((val) => {
+    if (val) {
+      return new Date(String(val)).toISOString();
+    }
+    return new Date().toISOString();
+  }, z.string())
```
Comment (thibaultleouay) [inline apps/server/src/v1/statusReportUpdate.ts]: We should extract that to 
 
```
const isoDate = z
  .preprocess((val) => {
    if (val) {
      return new Date(String(val)).toISOString();
    }
    return new Date().toISOString();
  }, z.string())
  ```
  Thus if we have a single source of truth

Comment (skushagra9) [reply apps/server/src/v1/statusReportUpdate.ts]: sure

Comment (skushagra9) [reply apps/server/src/v1/statusReportUpdate.ts]: should I store it in some other file , as its used in many places ??

File: apps/server/src/v1/statusReport.test.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { expect, test } from "bun:test";
-
 import { api } from ".";
 
+const iso8601Regex: RegExp = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z$/;
```
Comment (thibaultleouay) [inline apps/server/src/v1/statusReport.test.ts]: We should probably export in a file test-utils.ts

File: apps/server/src/v1/monitor.ts [hunk 3]
```diff
@@ -31,6 +31,14 @@ const ParamsSchema = z.object({
     }),
 });
 
+const isoDate = z
+  .preprocess((val) => {
```
Comment (thibaultleouay) [inline apps/server/src/v1/monitor.ts]: this should be exported in a `utils.ts` and import it where ever we need it

Comment (skushagra9) [reply apps/server/src/v1/monitor.ts]: sure

**Comment by skushagra9** [comment]
is it working now ?

**Review by thibaultleouay — APPROVED** [review]
🔥 Perfect

**Pull request merged**